compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11358,17 +11358,25 @@ private STNode parseTupleMemberTypeDescList() {
         }
 
         // Parse first typedesc, that has no leading comma
+        STNode annot = parseOptionalAnnotations();
         STNode typeDesc = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_TUPLE);
-
-        return parseTupleTypeMembers(typeDesc, typeDescList);
+        return parseTupleTypeMembers(annot, typeDesc, typeDescList);
     }
 
     private STNode parseTupleTypeMembers(STNode typeDesc, List<STNode> typeDescList) {
-        typeDesc = parseComplexTypeDescriptor(typeDesc, ParserRuleContext.TYPE_DESC_IN_TUPLE, false);
+        return parseTupleTypeMembers(STNodeFactory.createEmptyNode(), typeDesc, typeDescList);
+    }
 
+    private STNode parseTupleTypeMembers(STNode annot, STNode typeDesc, List<STNode> typeDescList) {
+        typeDesc = parseComplexTypeDescriptor(typeDesc, ParserRuleContext.TYPE_DESC_IN_TUPLE, false);
         STNode tupleMemberRhs = parseTypeDescInTupleRhs();
         if (tupleMemberRhs != null) {
+            if (annot != null) {
```

Comment:
Once `createEmptyNodeList()` is used as suggest above.
```suggestion
            if (!((STNodeList) annot).isEmpty()) {
```